cmd/query/app/apiv3/http_gateway.go

```diff
@@ -166,11 +160,24 @@ func (h *HTTPGateway) getTrace(w http.ResponseWriter, r *http.Request) {
 		}
 		request.RawTraces = rawTraces
 	}
-	trc, err := h.QueryService.GetTrace(r.Context(), request)
+	getTracesIter := h.QueryService.GetTraces(r.Context(), request)
+	trc, err := iter.FlattenWithErrors(getTracesIter)
```

Comment:
FlattenWithErrors still returns a collection, it doesn't combine everything into a single `Traces` object, which is what you need here.